src/python/pants/backend/project_info/rules/dependencies.py

```diff
@@ -51,43 +60,47 @@ class Dependencies(Goal):
 async def dependencies(
     console: Console, addresses: Addresses, options: DependenciesOptions,
 ) -> Dependencies:
-    address_strings: Set[str] = set()
-    third_party_requirements: Set[str] = set()
-
     if options.values.transitive:
-        transitive_targets = await Get[TransitiveHydratedTargets](Addresses, addresses)
-        hydrated_targets = HydratedTargets(
-            transitive_targets.closure - set(transitive_targets.roots)
-        )
+        transitive_targets = await Get[TransitiveTargets](Addresses, addresses)
+        targets = Targets(transitive_targets.closure - FrozenOrderedSet(transitive_targets.roots))
     else:
-        targets = await Get[HydratedTargets](Addresses, addresses)
-        direct_dependency_addresses = [
-            dep for target in targets for dep in target.adaptor.dependencies
-        ]
-        hydrated_targets = await Get[HydratedTargets](
-            Addresses, Addresses(direct_dependency_addresses)
+        target_roots = await Get[Targets](Addresses, addresses)
+        targets = await Get[Targets](
+            Addresses(
+                itertools.chain.from_iterable(
```

Comment:
Is using a nested for comprehension recommended over `from_iterable` these days...?

Reply:
I don't think so. I used to use nested for comprehensions to flatten, but @jsirois introduced me to `itertools.chain.from_iterable()` and I've come to prefer it because it's much easier to quickly know what this is doing. It also seems to work better with MyPy for some reason.

Still, I acknowledge it's not great. One of my biggest frustrations with Python is the lack of a `flatten()` built-in.